storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -79,11 +96,304 @@ public StormTopology getTopology() {
                 ret.put(executor, compId);
             }
         }
-        
+
         return ret;
     }
-    
+
     public Collection<ExecutorDetails> getExecutors() {
         return this.executorToComponent.keySet();
     }
+
+    private void initResourceList() {
+        _resourceList = new HashMap<ExecutorDetails, Map<String, Double>>();
+        // Extract bolt memory info
+        if (this.topology.get_bolts() != null) {
+            for (Map.Entry<String, Bolt> bolt : this.topology.get_bolts().entrySet()) {
+                //the json_conf is populated by TopologyBuilder (e.g. boltDeclarer.setMemoryLoad)
+                Map<String, Double> topology_resources = backtype.storm.scheduler.resource.Utils.parseResources(bolt
+                        .getValue().get_common().get_json_conf());
+                backtype.storm.scheduler.resource.Utils.checkIntialization(topology_resources, bolt.getValue().toString(), this.topologyConf);
+                for (Map.Entry<ExecutorDetails, String> anExecutorToComponent : executorToComponent.entrySet()) {
+                    if (bolt.getKey().equals(anExecutorToComponent.getValue())) {
+                        _resourceList.put(anExecutorToComponent.getKey(), topology_resources);
+                    }
+                }
+            }
+        } else {
+            LOG.warn("Topology " + topologyId + " does not seem to have any bolts!");
+        }
+        // Extract spout memory info
+        if (this.topology.get_spouts() != null) {
+            for (Map.Entry<String, SpoutSpec> spout : this.topology.get_spouts().entrySet()) {
+                Map<String, Double> topology_resources = backtype.storm.scheduler.resource.Utils.parseResources(spout
+                        .getValue().get_common().get_json_conf());
+                backtype.storm.scheduler.resource.Utils.checkIntialization(topology_resources, spout.getValue().toString(), this.topologyConf);
+                for (Map.Entry<ExecutorDetails, String> anExecutorToComponent : executorToComponent.entrySet()) {
+                    if (spout.getKey().equals(anExecutorToComponent.getValue())) {
+                        _resourceList.put(anExecutorToComponent.getKey(), topology_resources);
+                    }
+                }
+            }
+        } else {
+            LOG.warn("Topology " + topologyId + " does not seem to have any spouts!");
+        }
+        //schedule tasks that are not part of components returned from topology.get_spout or topology.getbolt (AKA sys tasks most specifically __acker tasks)
+        for(ExecutorDetails exec : this.getExecutors()) {
+            if (_resourceList.containsKey(exec) == false) {
+                LOG.info(
```

Comment:
Can we make this debug instead?  I don't think we need to tell everyone all the time about all of the topologies resources.